boat_nav/src/navigator_node.py

```diff
@@ -188,27 +189,24 @@ def vmg(direct_heading):
 def calc_global_max_vmg(wind_coming):
 	
 	# TODO: Make a service and make more general for object avoidance
-	upper_bound = wind_coming + layline
-	lower_bound = wind_coming - layline
+	upper_bound = angles.normalize(wind_coming + layline)
+	lower_bound = angles.normalize(wind_coming - layline)
+
 	# If direct heading is in irons, then max_vmg will be on the edge of the no go zone
-	if is_within_bounds(direct_heading, upper_bound, lower_bound):
-		theoretic_boat_speed = 0
+	if angles.is_within_bounds(direct_heading, lower_bound, upper_bound):
 
-		if gtAngle(direct_heading, wind_coming):
-			vmg_heading = (wind_coming + layline) % 360
-			theoretic_boat_speed = 2.5 * 0.514444
+		# Snap to whichever edge of the no go zone is closer
+		if angles.is_on_left(direct_heading, wind_coming):
+			vmg_heading = upper_bound
 		else:
-			vmg_heading = wind_coming - layline
-			if vmg_heading < 0:
-				vmg_heading += 360
-			theoretic_boat_speed = 2.5 * 0.514444
+			vmg_heading = lower_bound
 	
 	# Otherwise max vmg is just the direct heading
 	else:
-		theoretic_boat_speed = 2.5 * 0.514444 # 2.5 Knots to m/s (measured boat speed)
 		vmg_heading = direct_heading
 	
-	max_vmg = theoretic_boat_speed * math.cos(math.radians(vmg_heading - direct_heading))
+	theoretic_boat_speed = 2.5 * 0.514444 # 2.5 Knots to m/s (measured boat speed)
```

Comment:
Knots to m/s in utilities would be solid I think too.  Eventually I think we should have a `boat_model` module/class to represent the physics of the boat, so that `2.5` magic number can go in there.